src/rf/apps/home/views.py

```diff
@@ -199,13 +202,37 @@ def _get_layer_models(request, crit=None):
     if crit:
         qs = qs.filter(crit)
 
-    layers = LayerFilter(request.GET, queryset=qs)
-    return layers
+    filtered_layers = LayerFilter(request.GET, queryset=qs)
+    paginator = Paginator(filtered_layers, RESULTS_PER_PAGE)
+
+    page = request.GET.get('page')
+    try:
+        layers = paginator.page(page)
+    except PageNotAnInteger:
+        # If page is not an integer, deliver first page.
+        page = 1
+        layers = paginator.page(page)
+    except EmptyPage:
+        # If page is out of range, deliver last page of results.
+        page = paginator.num_pages
+        layers = paginator.page(page)
+
+    return {
+        'layers': layers,
+        'pages': paginator.num_pages,
+        'current_page': page
+    }
 
 
 def _get_layers(request, crit=None):
     """
     Return list of JSON serializable layer models.
     """
-    models = _get_layer_models(request, crit)
-    return [m.to_json() for m in models]
+    results = _get_layer_models(request, crit)
+    models = [m.to_json() for m in results['layers']]
+
+    return {
+        'layers': models,
+        'pages': results['pages'],
+        'current_page': results['current_page']
```

Comment:
I question the usefulness of numbered pages when searching, sorting, and filtering are available, but no harm in leaving as-is.